contracts/NodeOperatorRegistry.sol

```diff
@@ -1074,6 +1104,22 @@ contract NodeOperatorRegistry is
                 index++;
             }
         }
+
+        // Bug: Major 1.3 and 1.4
+        // If (index != totalActiveNodeOperator - 1) that mean there are empty fields
+        // in the array, it happens only in the case when a validator was unstaked
+        // directly from the stakeManager and the node operator registry not yet updated
+        // the operator status.
+        if (index != totalActiveNodeOperator - 1) {
```

Comment:
I'm going to need some clarification. 🤔

Reply:
Yes
Let's say we have 3 validators, then 1 validator was unstaked by the stakemanager.
In this case the nodeOperator contract we still have 3 active validator. SO wen we call getOperatorInfo we are creating an array of 3 validator. Then when we use this check (validator.status == IStakeManager.Status.Active) we are going to add only 2 validators to this array and the third one will be empty.
So, later this empty operator will revert when delegate, or distribute rewards.
This is why I added this additional loop to remove the empty operator.
Is that make sense for you?